RecoParticleFlow/PFClusterProducerAlpaka/plugins/alpaka/PFClusterParamsESProducer.cc

```diff
@@ -0,0 +1,254 @@
+#include "FWCore/Framework/interface/EventSetupRecordIntervalFinder.h"
+#include "FWCore/ParameterSet/interface/ConfigurationDescriptions.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/ParameterSet/interface/ParameterSetDescription.h"
+#include "FWCore/Utilities/interface/Exception.h"
+#include "HeterogeneousCore/AlpakaCore/interface/alpaka/ESProducer.h"
+#include "HeterogeneousCore/AlpakaCore/interface/alpaka/ModuleFactory.h"
+#include "HeterogeneousCore/AlpakaInterface/interface/config.h"
+#include "HeterogeneousCore/AlpakaInterface/interface/memory.h"
+#include "RecoParticleFlow/PFClusterProducerAlpaka/interface/PFClusterParamsAlpakaESRecord.h"
+#include "RecoParticleFlow/PFClusterProducerAlpaka/interface/alpaka/PFClusterParamsAlpakaESData.h"
+
+#include <array>
+
+namespace ALPAKA_ACCELERATOR_NAMESPACE {
+
+  class PFClusterParamsESProducer : public ESProducer {
+  public:
+    PFClusterParamsESProducer(edm::ParameterSet const& iConfig) : ESProducer(iConfig), paramSet(iConfig) {
+      product = std::make_shared<reco::PFClusterParamsAlpakaESDataHost>(7, cms::alpakatools::host());
+      constexpr static uint32_t kMaxDepth_barrel = 4;
+      constexpr static uint32_t kMaxDepth_endcap = 7;
+      auto view = product->view();
+
+      // seedFinder
+      auto const& sfConf = paramSet.getParameterSet("seedFinder");
+      view.nNeigh() = sfConf.getParameter<int>("nNeighbours");
+      auto const& seedFinderConfs = sfConf.getParameterSetVector("thresholdsByDetector");
+      for (auto const& pset : seedFinderConfs) {
+        auto const& det = pset.getParameter<std::string>("detector");
+        auto seedPt2Threshold = std::pow(pset.getParameter<double>("seedingThresholdPt"), 2.);
+        auto const& thresholds = pset.getParameter<std::vector<double>>("seedingThreshold");
+        if (det == "HCAL_BARREL1") {
+          if (thresholds.size() != kMaxDepth_barrel)
+            throw cms::Exception("InvalidConfiguration")
```

Comment:
The usual category for configuration errors is just
```suggestion
            throw cms::Exception("Configuration")
```